src/Core/Utility/ListEx.re

```diff
@@ -11,6 +11,23 @@ let rec last =
   | [x] => Some(x)
   | [_, ...t] => last(t);
 
+let mapResult: list(result('a, 'b)) => result(list('a), 'b) =
+  items => {
+    List.fold_left(
+      (acc, curr) => {
+        acc
+        |> ResultEx.flatMap(list => {
+             switch (curr) {
+             | Ok(v) => Ok([v, ...list])
+             | Error(_) as err => err
+             }
+           })
+      },
+      Ok([]),
+      items,
+    );
+  };
+
```

Comment:
This is a pretty bad name, as it doesn't actually do any mapping from he user POV. It also belongs in `ResultEx`, not `ListEx`.  It's a tricky one to name though. In Haskell and Scalaz it's called `sequence`, in Elm it's called `combine`, in ocaml-containers it's called `flatten` and in Base it's called `all`.

That last one should serve as a hint that you don't even have to implement and name it yourself, but can just use `Base.Result.all`.

Reply:
Ah, cool - I figured it must exist somewhere.  I keep forgetting that we have `Base` around... would be nice if there was something like `hoogle` for all our dependencies, to search by function signature.